src/connectors/tests/clickhouse.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// As we only have a sink implementation, we'll put everything here. Everything
+// will eventually follow a structure similar to the s3 connector.
+
+use crate::errors::{Error, Result};
+
+use std::time::Duration;
+
+use async_std::stream::StreamExt;
+use clickhouse_rs::Pool;
+use signal_hook::consts::{SIGINT, SIGQUIT, SIGTERM};
+use signal_hook_async_std::Signals;
+use testcontainers::{clients, core::Port, images::generic::GenericImage, Docker, RunArgs};
+use tremor_common::ports::IN;
+use tremor_pipeline::{Event, EventId};
+use tremor_script::literal;
+
+use super::ConnectorHarness;
+
+const CONTAINER_NAME: &str = "clickhouse/clickhouse-server";
+const CONTAINER_VERSION: &str = "22.3.3.44";
+const SERVER_PORT: u16 = 9000;
+
+const DB_HOST: &str = "127.0.0.1";
+
+#[async_std::test]
+async fn simple_insertion() -> Result<()> {
+    let _ = env_logger::try_init();
+
+    let docker = clients::Cli::default();
+    let image = GenericImage::new(format!("{CONTAINER_NAME}:{CONTAINER_VERSION}"));
+
+    // We want to access the container from the host, so we need to make the
+    // corresponding port available.
+    let port_to_expose = Port {
+        local: SERVER_PORT,
+        internal: SERVER_PORT,
+    };
+    let container =
+        docker.run_with_args(image, RunArgs::default().with_mapped_port(port_to_expose));
+
+    let container_id = container.id().to_string();
+    let mut signals = Signals::new(&[SIGTERM, SIGINT, SIGQUIT])?;
+    let signal_handle = signals.handle();
+    let signal_handler_task = async_std::task::spawn(async move {
+        let signal_docker = clients::Cli::default();
+        while let Some(_signal) = signals.next().await {
+            signal_docker.stop(container_id.as_str());
+            signal_docker.rm(container_id.as_str());
+        }
+    });
+    let port = container.get_host_port(9000).unwrap();
+
+    wait_for_ok(port).await.unwrap();
+
+    create_table(port, "people").await.unwrap();
+
+    let connector_config = literal!({
+        "config": {
+            "host": DB_HOST,
+            "port": port,
+            "database": "",
+        },
+    });
+
+    let harness = ConnectorHarness::new("clickhouse", &connector_config).await?;
+
+    let in_pipe = harness.get_pipe(IN).expect("No pipe connected to port IN");
+
+    harness.start().await?;
+    harness.wait_for_connected().await?;
+
```

Comment:
I would advise you to add a call to ```rust

Reply:
Fixed by [1187b8c](https://github.com/tremor-rs/tremor-runtime/pull/1538/commits/1187b8c29068f1325467137098fc89fbeeb14b7b)